src/ui/zcl_abapgit_gui_page_diff.clas.abap

```diff
@@ -1020,4 +1000,19 @@ CLASS zcl_abapgit_gui_page_diff IMPLEMENTATION.
     ENDCASE.
 
   ENDMETHOD.
+
+
+  METHOD is_file_requested.
+
+    IF lines( it_files ) = 0.
```

Comment:
I guess this IF is a suboptimization? READ TABLE for an empty table _should_ be as fast as an IF

Reply:
Well, wouldn't that change its semantics? READ TABLE for an empty table gives, if I'm not mistaken, SUBRC = 4 and then the method returns ABAP_FALSE.

Reply:
If it_files is empty, the original status loop was ok. If it's not empty, then I would do a loop with read table it_files inside.

Reply:
Do you mean something like this?
```
      IF lines( it_files ) = 0.

        LOOP AT lt_status ASSIGNING <ls_status> WHERE match IS INITIAL.

          append_diff( it_remote = lt_remote
                       it_local  = lt_local
                       is_status = <ls_status> ).

        ENDLOOP.

      ELSE.

        LOOP AT lt_status ASSIGNING <ls_status> WHERE match IS INITIAL.

          READ TABLE it_files WITH KEY file-path     = <ls_status>-path
                                       file-filename = <ls_status>-filename
                              TRANSPORTING NO FIELDS.
          IF sy-subrc = 0.
            append_diff( it_remote = lt_remote
                         it_local  = lt_local
                         is_status = <ls_status> ).
          ENDIF.

        ENDLOOP.

      ENDIF.

```

Reply:
Yes. Many ways to code the same thing but that looks good to me.

Reply:
ahh, okay, it says abap_true in line 1018